dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/deduplication/HibernatePotentialDuplicateStore.java

```diff
@@ -256,14 +256,18 @@ private void addTrackedEntityChangeLog(
 
   public void moveRelationships(
       TrackedEntity original, TrackedEntity duplicate, Set<UID> relationships) {
-    duplicate.getRelationshipItems().stream()
-        .filter(r -> relationships.contains(UID.of(r.getRelationship())))
-        .forEach(
-            ri -> {
-              ri.setTrackedEntity(original);
+    List<RelationshipItem> duplicateRelationshipItems =
```

Comment:
why is this change needed?

Reply:
The deduplication service is merging two TE in one, where the original will be kept and the duplicated removed.
In here, the `relationshipItem` was updated with the original `TE` to link to but then it wasn't moved to the original TE. This way, when deleting the duplicate TE, the required `relationshipItem` was deleted and the merge was not behaving as expected.

Basically, the merge was creating 2 `invalid` TE where the original was missing one `relationshipItem` and the duplicate was carrying one `relationshipItem` that didn't belong anymore to such TE.